UI/EncyclopediaDetailPanel.cpp

```diff
@@ -2428,7 +2428,28 @@ namespace {
         // show image of planet environment at the top of the suitability report
         std::string planet_type = boost::lexical_cast<std::string>(planet->Type());
         boost::algorithm::to_lower(planet_type);
-        detailed_description += "<img src=\"encyclopedia/planet_environments/" + planet_type + ".png\"></img>";
+
+        namespace fs = boost::filesystem;
+        std::vector<std::string> filenames;
+        fs::directory_iterator end_it;
```

Comment:
make `filenames` static, and wrap this all in `if (filenames.empty()) {` so that it only searches once per execution?

Reply:
Done (though it was a bit more complicated than just making it static, since it has to cache the data for every planet type).